New documentation using soss. [8217] (#14)
eProsima/Integration-Service-docs: docs/use_cases.rst

Use cases
=========

In this section the most representative use-cases are shown:

DDS Bridge
^^^^^^^^^^

> A typical scenario when communicating different `DDS` systems is to face that these systems use incompatible

Reviewer: ```suggestion
A typical scenario when communicating different `DDS` systems is to find out that these systems use incompatible
```

> configuration, for example, `DDS`'s HelloWorld and `ROS 2`'s Talker-Listener.

Reviewer: ```suggestion
configuration; for example, `DDS`'s HelloWorld and `ROS 2`'s Talker-Listener.
```

.. image:: DDS_NO_COMS.png

A user with knowledge of both systems may know that `ROS 2` uses `DDS` as middleware but hides some

> configuration details making difficult, or even impossible a direct communication.

Reviewer: ```suggestion
configuration details, making it difficult, or even impossible, to establish a direct communication.
```

By using `integration-service` this communication can be achieved with minimal effort from the user.

.. image:: DDS_WITH_IS.png

See :ref:`ROS2` for an example of this use-case.

Integrate a large system
^^^^^^^^^^^^^^^^^^^^^^^^

> Most systems evolve with the time, adding new functionality or new components.

Reviewer: ```suggestion
Most systems evolve over time, adding new functionality or new components.
```

When these new components are software or hardware, that doesn't use a protocol compatible with the rest
of the system, we are forced to create an additional component usually known as *bridge*.

> If the system contains several subsystems, and each component use a different protocol, a *bridge* must be
> created for each existing component pair that must communicate.

Reviewer: ```suggestion
If the system contains several subsystems, and each component uses a different protocol, a *bridge* must be
created for each existing component pair that must have communication capabilities.
```
